src/basic/pidref.h

```diff
@@ -1,12 +1,16 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 #pragma once
 
+#include <sys/types.h>
+
 #include "macro.h"
 
 /* An embeddable structure carrying a reference to a process. Supposed to be used when tracking processes continuously. */
 typedef struct PidRef {
-        pid_t pid; /* always valid */
-        int fd;    /* only valid if pidfd are available in the kernel, and we manage to get an fd */
+        pid_t pid;   /* always valid */
+        int fd;      /* only valid if pidfd are available in the kernel, and we manage to get an fd */
+        ino_t fd_id; /* the inode number of pidfd. only useful in kernel 6.9+ where pidfds live in
```

Comment:
hmm, ino_t is such a shit type, because it can be 32bit on some old systems.

Let's use uint64_t here always, on any system. i.e. just replace ino_t → uint64_t.

Reply:
that also means you can drop the sys/types.h include then